doc/conf.py

```diff
@@ -291,3 +291,12 @@ def setup(app):
                                          u'https://github.com/scikit-learn/'
                                          'scikit-learn/blob/{revision}/'
                                          '{package}/{path}#L{lineno}')
+
+
+# Configure custom autosummary to add a suffix to modules that overlap with
+# a upper case module, i.e. `sklearn.cluster.DBSCAN` and
+# `sklearn.cluster.dbscan`
+custom_autosummary_file_map = {
+    "sklearn.cluster.dbscan": "sklearn.cluster.dbscan_lowercase",
+    "sklearn.cluster.optics": "sklearn.cluster.optics_lowercase"
+}
```

Comment:
sklearn.covariance.oas and sklearn.decomposition.fastica need the same treatment